### PR TITLE
Update JSONCompareResult.java

### PR DESCRIPTION
**Pull request by riccorazza**
Improved classe adding 2 new lists for missing and unexpected fileds.

### CODE REVIEW

**Comment by carterpage** [comment]
Seems like an improvement.  @hertzsprung do you still watch this project?  Modulo indent fixes and adding a unit test, what do you think?

**Comment by dhawken** [comment]
Nice. Cloned to make this exact change.  +1 on merge and release.

**Comment by carterpage** [comment]
Okay, @riccorazza can you please add or update a few unit tests for this change?

File: src/main/java/org/skyscreamer/jsonassert/JSONCompareResult.java [hunk 1]
```diff
@@ -69,7 +85,8 @@ public String getMessage() {
      *         particular field
      * @deprecated Superseded by {@link #getFieldFailures()}
      */
-    public Object getActual() {
+    @Deprecated
+	public Object getActual() {
```
Comment (carterpage) [inline src/main/java/org/skyscreamer/jsonassert/JSONCompareResult.java]: Fix indentation.

File: src/main/java/org/skyscreamer/jsonassert/JSONCompareResult.java [hunk 2]
```diff
@@ -81,7 +98,8 @@ public Object getActual() {
      *         particular field
      * @deprecated Superseded by {@link #getFieldFailures()}
      */
-    public Object getExpected() {
+    @Deprecated
+	public Object getExpected() {
```
Comment (carterpage) [inline src/main/java/org/skyscreamer/jsonassert/JSONCompareResult.java]: Fix indentation.

File: src/main/java/org/skyscreamer/jsonassert/JSONCompareResult.java [hunk 3]
```diff
@@ -99,7 +131,8 @@ public boolean isFailureOnField() {
      *         not fail on a particular field
      * @deprecated Superseded by {@link #getFieldFailures()}
      */
-    public String getField() {
+    @Deprecated
+	public String getField() {
```
Comment (carterpage) [inline src/main/java/org/skyscreamer/jsonassert/JSONCompareResult.java]: Fix indentation.

**Comment by riccorazza** [comment]
Integrated tests in JSONCompareTest.java

**Comment by hertzsprung** [comment]
@carterpage Yes, I still lurk here though I'm no longer using Java or JSON :wink: Changes look fine to me, not sure why this didn't exist previously!  I'm a bit confused that @riccorazza made a comment about JSONCompareTest.java but the changes don't seem to be in this pull request.  Can you clarify, please?

**Comment by riccorazza** [comment]
But, how to add that changes here also?
Sorry, but I'm new on GITHUB.

**Comment by hertzsprung** [comment]
No problem @riccorazza.  If you make the changes in your branch (patch-1) they should be reflected in this pull request automatically.

**Comment by carterpage** [comment]
Ping @riccorazza.  Would love to get this in.  Any luck updating the PR?

**Comment by riccorazza** [comment]
Ok @hertzsprung, modified also tests.
I wait for the new PR.
Thanks

**Comment by riccorazza** [comment]
Are there any news?
I would like to you an updated package in my solution.
thanks

**Comment by carterpage** [comment]
Please fix the indentations per my comments, then ping this conversation, and I'll merge.

**Comment by riccorazza** [comment]
Indentation fixed...
Is everything ok now?
Thanks

**Comment by riccorazza** [comment]
When will it be available with maven?
Thaks

**Comment by carterpage** [comment]
I just noticed a few bugs (not introduced by you) that need fixing. I'm going to clean those up as I find time in the next week, then get it out to maven, which takes another week. So I'd estimate two weeks.

**Comment by riccorazza** [comment]
Is the new version ready to reference in Maven?
thanks

**Comment by carterpage** [comment]
Sorry I am so slow. I got slammed by my day job. I pushed the new version (1.3.0) to Maven today. It takes 24-48 hours to propagate to the repositories. Thanks for your contributions.

**Comment by drewhemm** [comment]
I'm comparing two objects and while the message says `data[]: Expected 2 values but got 1`, `getFieldMissing()` (along with `getFieldFailures()` and ` getFieldUnexpected()`) returns an empty list - is this expected behaviour in 1.5.0?

**Pull request merged**